### ISSUE TITLE
Style guidelines

### ISSUE THREAD

**Issue by stil4m** [stil4m/mollie-api#28]
There should be style guidelines for the Java code to minimize the changes.

These should at least address:
- Import order
- Use of `*` imports
- Amount of whitespace between methods and fields.

Should the repo contain a settings file/JAR, or just some guidelines?

**Comment by tubbynl** [comment]
I agree, i tried to keep it to a minimum (disable save-actions formatting in eclipse).

What IDE do you use because i am considering switching IDE?

**Comment by tubbynl** [comment]
btw; a bit offtopic but can i expect @stil4m or @jcassee at jfall?

**Comment by stil4m** [comment]
Nope. Study and work have priority at this moment.

**Comment by jcassee** [comment]
@tubbynl Sorry, not this year.

**Comment by tubbynl** [comment]
your whitespace settings are?
- Spaces only
- Indentation/tab size: 2

or perhaps we could conform to some styleguide already out there; example: https://google.github.io/styleguide/javaguide.html

i'm not really specific on these things just that it's practical to be in sync in order to prevent unneeded merge conflicts.

**Comment by stil4m** [comment]
I think I have it on 4 spaces. I want to cover this with a checkstyle config, which will run with CI. I already have a local branch for this.

I'm not sure if we should take all the guidelines or differ on some. 100 columns seem a bit harsh IMHO.
Other file types such as JSON are not addressed, but need to be accounted for as well. 

I'm kinda swamped ATM with work and a master program, thus this will not have much prio at the moment. For now I will run a formatter after each release manually (most optimal with the available time/time to fix this ATM).

**Comment by tubbynl** [comment]
the example was more of an indication; i mostly try to use the default Java conventions but with some changes (200 line width and space-only indentation on 4 characters).

i've already set auto-formatting off on this project so it doesn't interfere too much :)

tnx for merging my pull-requests btw :)

**Comment by stil4m** [comment]
Not a optimal solution, but this is a link to my `Code Style` settings from Idea: https://cloudup.com/cG-oeKjS5uC

Maybe you can enable it for this project (for the time being)

**Comment by tubbynl** [comment]
for the record; java-wise your CodeStyle settings only specify the import-counts to start using .\* imports both on 99

the rest is standard IntelliJ (also the import order differ for IntelliJ and Eclipse sadly :/ )

**Comment by jcassee** [comment]
@tubbynl It is possible to instruct IntelliJ to use Eclipse's import order, I think.

**Comment by tubbynl** [comment]
The problem was the other way around (me using eclipse ordering imports differently)

**Comment by stil4m** [comment]
Does anybody have experience with [this library](https://github.com/google/google-java-format) or something similar.

I've been using [elm-format](https://github.com/avh4/elm-format) extensively and I like the simplicity of 'I do not bother with the style, it just looks nice'.

**Comment by tubbynl** [comment]
nope,

but if you currently use elm-format i think that makes the most sense, i mostly think collaboration/less diffs is more important than some-person-s-code-style-preference :)
